src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/impl/ActionSvcRestClientImpl.java

```diff
@@ -97,35 +109,62 @@ public ActionPlanDTO createActionPlan(
         collectionExerciseId,
         activeEnrolment);
 
-    MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
-    queryParams.add("collectionExerciseId", collectionExerciseId);
-    queryParams.add("activeEnrolment", activeEnrolment.toString());
-    UriComponents uriComponents =
-        restUtility.createUriComponents(appConfig.getActionSvc().getActionPlansPath(), queryParams);
-
-    ResponseEntity<List<ActionPlanDTO>> responseEntity;
-    try {
-      responseEntity =
-          restTemplate.exchange(
-              uriComponents.toString(),
-              HttpMethod.GET,
-              null,
-              new ParameterizedTypeReference<List<ActionPlanDTO>>() {});
-    } catch (HttpClientErrorException e) {
-      if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
-        return null;
-      }
-      throw e;
-    }
+    final MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
+    queryParams.add(SELECTOR_COLLECTION_EXERCISE_ID, collectionExerciseId);
+    queryParams.add(SELECTOR_ACTIVE_ENROLMENT, activeEnrolment.toString());
+    final ResponseEntity<List<ActionPlanDTO>> responseEntity =
+        getActionPlansByQueryParameters(queryParams);
 
     log.debug(
         "Successfully retrieved action plan for selectors, "
             + "collectionExerciseId: {}, activeEnrolment: {}",
         collectionExerciseId,
         activeEnrolment);
+
     return responseEntity.getBody();
   }
 
+  private ResponseEntity<List<ActionPlanDTO>> getActionPlansByQueryParameters(
```

Comment:
this is inconsistent with the other getters which return a list. Consider removing the ResponseEntity 'wrapper' and just returning the list inside

Reply:
No longer needed, inlined back into getActionPlansBySelectors